src/reactviews/pages/ConnectionGroup/connectionGroup.component.tsx

```diff
@@ -0,0 +1,253 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useFormStyles } from "../../common/forms/form.component";
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogBody,
+    DialogContent,
+    DialogSurface,
+    DialogTitle,
+    Input,
+    Textarea,
+    MessageBar,
+    Field,
+    makeStyles,
+    Popover,
+    PopoverTrigger,
+    PopoverSurface,
+} from "@fluentui/react-components";
+import {
+    ColorArea,
+    ColorPicker,
+    ColorPickerProps,
+    ColorSlider,
+} from "@fluentui/react-color-picker";
+import { TinyColor } from "@ctrl/tinycolor";
+import { locConstants as Loc } from "../../common/locConstants";
+import {
+    ConnectionGroupSpec,
+    ConnectionGroupState,
+} from "../../../sharedInterfaces/connectionGroup";
+import { useState } from "react";
+import { Keys } from "../../common/keys";
+
+const useStyles = makeStyles({
+    previewColor: {
+        width: "80px",
+        borderRadius: "4px",
+        border: "1px solid #ccc",
+        alignSelf: "stretch",
+        "@media (forced-colors: active)": {
+            forcedColorAdjust: "none",
+        },
+    },
+    row: {
+        display: "flex",
+        gap: "10px",
+    },
+    sliders: {
+        display: "flex",
+        flexDirection: "column",
+    },
+    colorContainer: {
+        display: "flex",
+        alignItems: "center",
+        gap: "15px",
+    },
+});
+
+/** Generates a random Hex color */
+function getRandomColor(): string {

```

Comment:
Should we just limit to specific colors that looks good. Something like a collection of flat fluent colors.

Reply:
I'll update this to pull from a set of known good fluent colors, but I'll admit that I was surprised in testing this function that it didn't tend to spit out "ugly" colors with particular frequency.